### PR TITLE
tinc: rework

### PR DESCRIPTION
**Pull request by awiouy**
I have reworked tinc.

Configuration is easier:
- less settings
- setting up port forwarding is not necessary if router supports UPnP
- clearer folder names in ADDON_HOME
- one host always corresponds to a single VPN IP address, as opposed to a subnet

Security is improved:
- tinc only allows connections with nodes for which host files are present (TunnelServer=yes)
- keys are less accessible
- tinc only routes unicast IP packets (Mode=Router) 

Purpose:
The tinc addon enables to connect LE devices to each other or to other tinc hosts through Internet

How it works:
1. Alice and Bob agree on VPN IP addresses ,10.11.12.13 and 10.9.8.7
2. Alice:
   a. sets up a DDNS name (alice.wonderland.net)
   b. configures tinc (Subnet=10.11.12.13, Address=alice.wonderland.net)
   c. sends the host file in ADDON_HOME/Local (10_11_12_13) to Bob
3. Bob:
   a. configures tinc (Subnet=10.9.8.7, no Address)
   b. stores Alice's host file (10_11_12_13) in ADDON_HOME/Remotes
   c. restarts tinc (eg, reboot)
   d. sends the host file in ADDON_HOME/Local (10_9_8_7) to Alice
4. Alice:
   a. stores Bob's host file (10_9_8_7) in ADDON_HOME/Remotes
   b. restarts tinc (eg, reboot)
5. Alice should see Bob (ping 10.9.8.7) and Bob should see Alice (ping 10.11.12.13)

Optional:
- Alice may have to set up port forwarding on her router if it does not support UPnP
- Before sending his host file, Bob may set up a DDNS name and set the corresponding Address to improve availability and connect time
- Alice or Bob can add further remote hosts, tinc will only allow connections with nodes for which host files are present in ADDON_HOME/Remotes

Warning:
With tinc on LibreELEC you give remote hosts root access to a system in your network. Uses other than personal are [irresponsible](http://www.thesaurus.com/browse/foolish)

### CODE REVIEW

File: packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#  along with LibreELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+PKG_NAME="miniupnpc"
+PKG_VERSION="2.0.20180222"
+PKG_SHA256="587944686469d09f739744b3aed70c7ce753a79c40d6f1227f68a3e962665b75"
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
+PKG_SITE="http://miniupnp.free.fr"
+PKG_URL="http://miniupnp.free.fr/files/$PKG_NAME-$PKG_VERSION.tar.gz"
```
Comment (CvH) [inline packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk]: maybe better use https://github.com/miniupnp/miniupnp as they doesn't have any proper releases anyway

File: packages/addons/service/tinc/changelog.txt [hunk 2]
```diff
@@ -1,5 +1,12 @@
-8.0.101
-- Bump to tinc-1.1pre14
+103
+- Update to tinc-1.1pre15
+- Radical rework
```
Comment (CvH) [inline packages/addons/service/tinc/changelog.txt]: `- rework` should be enough ;)

File: packages/addons/service/tinc/source/resources/language/English/strings.po [hunk 3]
```diff
@@ -1,70 +1,29 @@
 # Kodi Media Center language file
 # Addon Name: tinc
 # Addon id: service.system.tinc
-# Addon Provider: awiouy at gmail dot com
```
Comment (CvH) [inline packages/addons/service/tinc/source/resources/language/English/strings.po]: `# Addon Provider: Team LibreELEC` to keep in sync with the "standard" (that get hopefully merged some day)

File: packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#
+#  You should have received a copy of the GNU General Public License
+#  along with LibreELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+PKG_NAME="miniupnpc"
+PKG_VERSION="2.0.20180222"
+PKG_SHA256="587944686469d09f739744b3aed70c7ce753a79c40d6f1227f68a3e962665b75"
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
```
Comment (CvH) [inline packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk]: likely no GPL, not sure what it is but no gpl

File: packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk (hunk 1 above)
Comment (awiouy) [reply packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk]: I tried, but github ships both client and daemon in subdirectories. No idea how to handle that with the LE build system.

File: packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk (hunk 4 above)
Comment (awiouy) [reply packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk]: meh, it is BSD

File: packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk (hunk 1 above)
Comment (CvH) [reply packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk]: untested but should work this way, tbh not really a benefit in doing so
```
unpack() {
  mkdir -p $BUILD
    tar -C $BUILD/ --strip-components=1 -xzvf $SOURCES/$PKG_NAME/$PKG_NAME-$PKG_VERSION.tar.gz miniupnpc/
}
```

**maybe** (not tested at all) `PKG_SOURCE_DIR="${PKG_NAME}-${PKG_VERSION}/miniupnpc"`

Comment (awiouy) [reply packages/addons/addon-depends/tinc-depends/miniupnpc/package.mk]: Changing this does not make much sense at the moment, no?

File: packages/addons/service/tinc/source/resources/language/English/strings.po (hunk 3 above)
Comment (awiouy) [reply packages/addons/service/tinc/source/resources/language/English/strings.po]: Done

**Comment by AvverbioPronome** [comment]
Guys, I haven't been able to find any documentation on how to connect this to an existing tinc network (ie: where the network and hosts directories are, where the public key is, where to put the keys for other nodes).

Any help?

**Comment by CvH** [comment]
@AvverbioPronome pls open a thread at https://forum.libreelec.tv/

**Pull request merged**